content/specs/cryptography.md

```diff
@@ -32,6 +32,28 @@ In atproto, use of the "low-S" signature variant is required for both `p256` and
 In atproto, signatures should always be verified using the verification routines provided by the cryptographic library, never by comparing signature values as raw bytes.
 
 
-### Possible Future Changes
+## Public Key Encoding
+
+When encoding public keys as strings, the preferred representation uses multibase (with `base58btc` specifically) and a multicode prefix to indicate the specific key type. By embedding metadata about the type of key in the encoding itself, they can be parsed unambiguously. The process for encoding a public key in this format is:
+
+- Encode the public key curve "point" as bytes. Be sure to use the smaller "compact" or "compressed" representation. This is usually easy for `k256`, but might require a special argument or configuration for `p256` keys
+- Prepend the appropriate curve multicodec value, as varint-encoded bytes, in front of the key bytes:
+    - `p256` (compressed, 33 byte key length): `p256-pub`, code 0x1200, varint-encoded bytes: [0x80, 0x24]
+    - `k256` (compressed, 33 byte key length): `secp256k1-pub`, code 0xE7, varint bytes: [0xE7, 0x01]
+- Encode the combined bytes with `base58btc`, yielding a string
```

Comment:
and prefix with `z`

Reply:
good catch. also added example encoded keys just below